config/configvalidator.php

```diff
@@ -48,32 +48,18 @@ public function isConfigSafe($key, $parsedConfigItem) {
 	 *
 	 * @param array $parsedConfigItem the sorting configuration to analyse
 	 * @param bool $safe whether the current config has been deemed safe to use so far
-	 *
+	 * @param string key  whether this exist or not 
 	 * @return bool
 	 */
-	private function isSortingTypeSafe($parsedConfigItem, $safe) {
-		if ($safe && array_key_exists('type', $parsedConfigItem)) {
-			$safe = $safe && $this->sortingValidator('type', $parsedConfigItem['type']);
+	private function isSortingSafe($key,$parsedConfigItem, $safe) {
+		if ($safe && array_key_exists($key, $parsedConfigItem)) {
+			$safe = $safe && $this->sortingValidator($key, $parsedConfigItem[ $key ]);
 		}
 
 		return $safe;
 	}
 
-	/**
-	 * Determines if the sorting order found in the config file is safe for web use
-	 *
-	 * @param array $parsedConfigItem the sorting configuration to analyse
-	 * @param bool $safe whether the current config has been deemed safe to use so far
-	 *
-	 * @return bool
-	 */
-	private function isSortingOrderSafe($parsedConfigItem, $safe) {
-		if ($safe && array_key_exists('order', $parsedConfigItem)) {
-			$safe = $safe && $this->sortingValidator('order', $parsedConfigItem['order']);
-		}
-
-		return $safe;
-	}
+	
```

Comment:
sorry , i will  definitely do better next time . learn a lot from this and thanks for your valuable suggestions :)